From: Jeroen Dekkers <jeroen@dekkers.ch>
Subject: Add so version number to private libraries for dpkg-shlibdeps
Origin: vendor
Forwarded: not-needed

We also want dpkg-shlibdeps to generate correct dependency information
for the private libraries in our binary packages, but dpkg-shlibdeps
only works when the library has a version number.

Index: samba/buildtools/wafsamba/wafsamba.py
===================================================================
--- samba.orig/buildtools/wafsamba/wafsamba.py
+++ samba/buildtools/wafsamba/wafsamba.py
@@ -232,6 +232,9 @@ def SAMBA_LIBRARY(bld, libname, source,
             raise Utils.WafError("public library '%s' must have header files" %
                        libname)
 
+    if private_library and not vnum:
+        vnum = '0'
+
     if bundled_name is not None:
         pass
     elif target_type == 'PYTHON' or realname or not private_library:
